ssl/t1_lib.c

```diff
@@ -577,19 +578,26 @@ void tls1_get_formatlist(SSL *s, const unsigned char **pformats,
 /* Check a key is compatible with compression extension */
 static int tls1_check_pkey_comp(SSL *s, EVP_PKEY *pkey)
 {
-    const EC_KEY *ec;
-    const EC_GROUP *grp;
     unsigned char comp_id;
     size_t i;
 
+    /*
+     * TODO(3.0) Remove this when we have routines that can extract the
+     * compression information from a provider side key.  We know that
+     * EVP_PKEY_get0() downgrades an EVP_PKEY to contain a legacy key.
+     *
+     * THIS IS TEMPORARY
+     */
+    EVP_PKEY_get0(pkey);
+    if (EVP_PKEY_id(pkey) == EVP_PKEY_NONE)
+        return 0;
```

Comment:
Now that we have downgraded we can keep all the original code below. The `EVP_PKEY_get0_EC_KEY()` call will just work and we don't need `evp_pkey_get_EC_KEY_conv_form` or `evp_pkey_get_EC_KEY_field_type`.

Reply:
At *some* point, we will need to get those things from provider side keys, so I'm apprehensive at removing those functions...  do you anticipate that the *final solution* will look substantially different?

Reply:
Ping @mattcaswell

Reply:
Yes. This whole dance with these calls is just to figure out what TLS compression id we need to use. I anticipate asking the provider directly for this compression id rather than inferring it from these other calls.

Reply:
(I assume "TLS compression id" means "EC point format", not "we're using something other than NULL compression".)

Reply:
Yes...the TLS EC point format. Sorry for the confusion.

Reply:
In that case, I might as well restore this whole function to its original form, and just replace the EVP_PKEY_id() check with a EVP_PKEY_is_a() check.

Reply:
Done